### DISCUSSION TITLE
TEM image segmentation issues

### DISCUSSION THREAD

**Discussion by JennyVenti** [neuropoly/axondeepseg#810] [Help needed]
Dear AxonDeepSeg community, 
I would to automatically segment peripheral nerves in order to look at the effect of different radiation treatment on myelinated axons. I tested several parameters (zoom factor and with/without patches) and resolution sizes (downsampled imaged with pre-processing). Here is the best segmentation I was able to get when using a 0.4 zoom factor with patch. 
The original image is: 3200x2622, 0.005912 um/pixel, 8 bit (tiff). I an running on a Windows 11, Intel(R) Core(TM) i7-10510U CPU. 
I have read a similar post for which you suggested going with a low zoom factor, which seems to be helping too in my case but I can not go lower than 0.4.
Do you think it is possible to improve the segmentation or should I train the model with my own dataset images?
Many thanks for your help,
Jennifer 

![Screenshot_0_059umPix_Patch_zoom0_4](https://github.com/axondeepseg/axondeepseg/assets/173394797/ac6fd570-61e0-4906-8deb-889d72beae45)

**Comment by Stoyan-I-A** [comment]
Hello @JennyVenti, 
Just for your information, we are currently working on training a "generalist" model that would not require a pixel size or zoom factor to properly segment an image. I'll tag @hermancollin if he wants to chip in on that.

Now about the zoom factor, the appropriate zoom factor to use depends a lot on what your pixel size (in um/pixel) is compared to what was the pixel size of the data used to train the model you are using. The pixel size of the models is described [here](https://axondeepseg.readthedocs.io/en/latest/documentation.html#existing-models) in our documentation. 
The dimensions of your image will be resampled using this formula:
`resampled_dimension = original_dimension*(acquired_resolution*zoom_factor)/model_resolution`
Ideally, the ressampled dimensions should not differ too much from the original dimension because otherwise the resampled image may look very different from the data the model was trained on.
This brings me to my last point, you could try different models. Ideally, you should pick a model that was trained on data that looks similar to your images.

**Reply by JennyVenti** [reply]
Hi @Stoyan-I-A,
Thank you so much for your responsiveness and explanation. I understand better now my issue due to different original pixel size acquired. Your suggestion would be to pick a model that was trained on data similar to mine. Can I use Ivadomed to create a model using manual ground truth segmentation on my own dataset?

**Reply by jcohenadad** [reply]
Hi @JennyVenti,

Thank you for reaching out. [ivadomed](https://ivadomed.org) project is not maintained anymore, and we are currently in the process of converting our training workflow to MONAI. Moreover, we are currently training a 'generalist' model that is being trained on a large variety of microscopy contrasts. If you send us 2-3 images, we could incorporate them into the model, and hopefully it will produce good results for you.

**Reply by JennyVenti** [reply]
Hi @jcohenadad,
That's good to know, this is going to be very helpful! I am new to TEM but it seems like I can see differences in contrast across imaging sessions. Should I email the images to axondeepseg@googlegroups.com or a different email?

**Reply by jcohenadad** [reply]
Yes, that email works 😊
